crates/core/tedge_api/src/health.rs

```diff
@@ -92,14 +96,195 @@ impl ServiceHealthTopic {
     }
 }
 
+#[derive(Deserialize, Serialize, Debug, Default)]
+pub struct HealthStatus {
+    #[serde(default)]
+    pub status: Status,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum Status {
+    Up,
+    Down,
+    #[serde(untagged)]
+    Other(String),
+}
+
+impl Default for Status {
+    fn default() -> Self {
+        Status::Other("unknown".to_string())
+    }
+}
+
+impl Display for Status {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let status = match self {
+            Status::Up => "up",
+            Status::Down => "down",
+            Status::Other(val) if val.is_empty() => "unknown",
+            Status::Other(val) => val,
+        };
+        write!(f, "{}", status)
+    }
+}
+
 #[derive(Debug)]
 pub struct HealthTopicError;
 
+impl HealthStatus {
+    pub fn try_from_health_status_message(
+        message: &MqttMessage,
+        mqtt_schema: &MqttSchema,
+        bridge_service_topic: &Topic,
```

Comment:
Having to take this `bridge_service_topic` is weird, but unfortunately unavoidable as the `builtin-bridge` also uses the mosquitto-style of payloads (`1` and `0` for backward compatibility, I assume), but does not conform to the mosquitto-bridge naming convention (starting with `mosquitto-` and ending with `-bridge`). So, we need to have special handling of the health status message from that specific service.